src/URI.js

```diff
@@ -1863,6 +1863,7 @@
 
     if (resolved.path().charAt(0) !== '/') {
       basedir = base.directory();
+      basedir = basedir ? basedir : base.path().indexOf('/') === 0 ? '/' : '';
```

Comment:
that reads horribly (not that my code reads any better…)
wondering if the following makes more sense

```js
// either basedir is a thing, or path begins with /, defaults to empty string
basedir || base.path().slice(0, 1) === '/' && '/' || '';
```